app/(main)/league/all/page.test.tsx

```diff
@@ -1,124 +1,295 @@
+        leagues: [],
+      },
+      allLeagues: [],
+    });
+
     render(<Leagues />);
 
+    await waitForElementToBeRemoved(() => screen.getByTestId('global-spinner'));
+
     await waitFor(() => {
-      expect(
-        screen.getByText('You are not enrolled in any leagues'),
-      ).toBeInTheDocument();
+      const messageElement = screen.getByTestId('no-leagues-message');
+      expect(messageElement).toBeInTheDocument();
     });
   });
 
-  xtest('should display GlobalSpinner while loading data', async () => {
-    mockUseDataStore.mockReturnValueOnce({ user: { id: '123', leagues: [] } });
-    mockGetUserLeagues.mockResolvedValueOnce([]);
-    mockGetGameWeek.mockResolvedValueOnce({ week: 1 });
-    render(<Leagues />);
+  test('should display GlobalSpinner while loading data', async () => {
+    mockUseAuthContext.isSignedIn = true;
 
-    await waitFor(() => {
-      expect(screen.getByTestId('global-spinner')).toBeInTheDocument();
+    mockUseDataStore.mockReturnValueOnce({
+      user: {
+        documentId: '123',
+        email: 'test@test.com',
+        id: '123',
+        leagues: [],
+      },
+      allLeagues: [],
     });
-  });
-  xtest('should not display GlobalSpinner after loading data', async () => {
-    mockUseDataStore.mockReturnValueOnce({ user: { id: '123', leagues: [] } });
-    mockGetUserLeagues.mockResolvedValueOnce([]);
-    mockGetGameWeek.mockResolvedValueOnce({ week: 1 });
 
     render(<Leagues />);
 
     expect(screen.getByTestId('global-spinner')).toBeInTheDocument();
-
-    await waitFor(() => {
-      expect(screen.queryByTestId('global-spinner')).not.toBeInTheDocument();
-    });
   });
 
-  xtest('should handle form submission to join a league', async () => {
-    mockUseDataStore.mockReturnValueOnce({
-      user: { id: '123', leagues: [] },
+  test('should not display GlobalSpinner after loading data', async () => {
+    mockUseAuthContext.isSignedIn = true;
+
+    mockUseDataStore.mockReturnValue({
+      user: {
+        documentId: '123',
+        email: 'test@test.com',
+        id: '123',
+        leagues: [],
+      },
       allLeagues: [
         {
           leagueId: '123',
           leagueName: 'Test League',
-          logo: 'https://findmylogo.com/logo.png',
+          logo: 'logo.png',
           participants: ['123456', '78'],
-          survivors: ['123456', '78', '9'],
+          survivors: ['123456', '78'],
         },
       ],
     });
+
     mockGetUserLeagues.mockResolvedValueOnce([]);
-    mockGetGameWeek.mockResolvedValueOnce({ week: 1 });
+
+    render(<Leagues />);
+
+    await waitForElementToBeRemoved(() => screen.getByTestId('global-spinner'));
+
+    expect(screen.queryByTestId('global-spinner')).not.toBeInTheDocument();
+  });
+
+  test('should handle form submission to join a league', async () => {
+    mockUseAuthContext.isSignedIn = true;
+
+    const user = {
+      documentId: '123',
+      email: 'test@test.com',
+      id: '123',
+      leagues: ['12'],
```

Comment:
this should be an empty array.  for this test you are wanting to say the user is not a part of any leagues.

Reply:
Updated